hw_3/script.js

```diff
@@ -0,0 +1,111 @@
+function checkArgLength(args) {
+    if (args.length !== 2) {
+        throw new Error("Not 2 arguments");
+    }
+}
+
+function checkMinMax(min, max) {
+    if (!Number.isInteger(min) || !Number.isInteger(max)) {
+        throw new Error("arguments are not Integer");
+    }
+    if (min > max) {
+        throw new Error("Min > Max");
+    }
+}
+
+function checkResult(result) {
+    if (result > Number.MAX_SAFE_INTEGER) {
+        return new Error("The result is greater than Number.MAX_SAFE_INTEGER")
+    } else {
+        return result;
+    }
+
+}
+
+function range(min, max) {
+    checkMinMax(min, max);
+    return checkResult((min + max) * (max - min + 1) / 2);
+}
+
+const memoize = (func) => {
+    let cache = {};
+    return (...args) => {
+        checkArgLength(args);
+        let n = JSON.stringify(args);
+        if (n in cache) {
+            console.log('Fetching from cache:');
+        } else {
+            let result = func(...args);
+            console.log('Result was calculated:');
+            cache[n] = result;
+        }
+        if (cache[n] instanceof Error) {
+            throw cache[n];
+        }
+        return cache[n];
+    }
+}
+
+const memoizedRange = memoize(range);
+try {
```

Comment:
It would be better if this check code were commented out and placed at the end.
```suggestion
// Code for testing error cases
/*
try { ...
*/
```

Reply:
Commented error checking.
https://github.com/BlackTruth/javascript/blob/813db56a38899f41dcfdc4b332943bde5c271592/hw_3/script.js#L56-L59